pgjdbc/src/main/java/org/postgresql/core/v2/ProtocolConnectionImpl.java

```diff
@@ -197,6 +209,15 @@ synchronized void addWarning(SQLWarning newWarning) {
 
   synchronized void addNotification(PGNotification notification) {
     notifications.add(notification);
```

Comment:
Have you considered how the notifications list gets cleared if a user is strictly relying upon the listener API?